cdap-api/src/main/java/io/cdap/cdap/api/metadata/MetadataEntity.java

```diff
@@ -175,8 +175,8 @@ public MetadataEntity build() {
         throw new IllegalArgumentException("key-value pair must be specified");
       }
       if (type == null) {
-        // traverse till the last key and make that the type
-        parts.keySet().forEach(x -> type = x);
+        // traverse till the last key except version and make that the type
+        parts.keySet().stream().filter(key -> !key.equals(VERSION)).forEach(x -> type = x);
```

Comment:
version cannot be type

Reply:
No need to change this, for custom entity we don't care about the type, just let the entity keeps the old behavior